packages/database/src/analytics/AnalyticsRefresher.js

```diff
@@ -0,0 +1,60 @@
+/**
+ * Tupaia
+ * Copyright (c) 2017 - 2021 Beyond Essential Systems Pty Ltd
+ */
+
+const REFRESH_DEBOUNCE_TIME = 1000; // wait 1 second after changes before refreshing, to avoid double-up
+
+export class AnalyticsRefresher {
+  constructor(database, models) {
+    this.database = database;
+    this.models = models;
+    this.scheduledRefreshTimeout = null;
+    this.scheduledRefreshPromise = null;
+    this.scheduledRefreshPromiseResolve = null;
+    this.changeHandlerCancellers = [];
+  }
+
+  listenForChanges() {
+    this.changeHandlerCancellers = [this.models.answer.addChangeHandler(this.handleAnswerChange)];
```

Comment:
Not a silly question at all, very relevant! I basically didn't wanna add too many listeners to avoid adding too much of a performance overhead. My logic is:
- If entities/questions/surveys/etc. are added, we don't need to update analytics until new answers are actually added to the system
- It's pretty rare for us to edit existing data_element_codes/entity_codes/etc. if so we can manually run a refresh? You should be able to do this via a migration

Valid point tho, what are your thoughts on my reasoning?

Reply:
I have to agree!